export/mlab_export.py

```diff
@@ -0,0 +1,605 @@
+
+  Returns:
+    dict, keys are raw metric names, values are canonical metric names.
+  Exits:
+    When filename is missing, has bad configuration, or is missing metrics section.
+  """
+
+  # ConfigParser.read ignores non-existent files, so check that the file exists.
+  if not os.path.exists(filename):
+    logging.error('Config file does not exist: %s', filename)
+    sys.exit(1)
+
+  try:
+    # Catch parsing or format errors.
+    parser = ConfigParser.SafeConfigParser()
+    parser.read(filename)
+  except ConfigParser.Error as err:
+    logging.error('Error while reading %s: %s', filename, err)
+    sys.exit(1)
+
+  if parser.has_section('metrics'):
+    metric_map = dict(parser.items('metrics'))
+  else:
+    logging.error('Config file is missing "[metrics]" section')
+    sys.exit(1)
+
+  return metric_map
+
+
+def any_show_options(options):
+  """Checks if any show options are True."""
+  return any([options.show_nagios, options.show_rrdfile, options.show_rawmetric,
+              options.show_metric, options.show_skipped])
+
+
+def init_args(options, ts_previous):
+  """Initializes flags with default values and asserts sanity checks.
+
+  Args:
+    options: flags.FlagValues, the unprocessed defaults from flags.FLAGS.
+    ts_previous: int, timestamp in seconds since epoch of last export.
+  Returns:
+    flags.FlagValues, options with updated defaults.
+  """
+  global METRIC_MAP
+  METRIC_MAP = read_metric_map(options.export_metrics)
+
+  if options.verbose:
+    logging.basicConfig(level=logging.DEBUG)
+
+  if options.ts_start is None:
+    options.ts_start = default_start_time(options, ts_previous)
+
+  if options.ts_end is None:
+    options.ts_end = (
+        align_timestamp(options.ts_start, options.step) + options.length)
+
+  if any_show_options(options):
+    options.update = False
+
+  if options.rrddir_prefix[-1] != os.path.sep:
+    # Ensure that the last character of rrddir_prefix includes path separator.
+    options.rrddir_prefix += os.path.sep
+
+  assert_start_and_end_times(options)
+
+  if options.output is None:
+    options.output = default_output_name(
+        options.ts_start, options.ts_end, options.output_dir)
+
+  return options
+
+
+def parse_args(ts_previous):
+  """Parses command line arguments and initialize defaults.
+
+  Args:
+    ts_previous: int, timestamp in seconds since epoch of last successful
+        export. On first run, this value should be zero.
+  Returns:
+    flags.FlagValues, all options.
+  """
+  try:
+    _ = flags.FLAGS(sys.argv)  # Parses flags. Any remaining args are unused.
+  except flags.FlagsError, err:
+    logging.error('%s\nUsage: %s ARGS\n%s', err, sys.argv[0], flags.FLAGS)
+    sys.exit(1)
+  try:
+    return init_args(flags.FLAGS, ts_previous)
+  except TimeOptionError as err:
+    logging.error(err)
+    sys.exit(1)
+
+
+def main():
+  init_global()
+
+  try:
+    lock_file = LockFile(EXPORT_LOCKFILE)
+    lock_file.acquire()
+  except IOError as err:
```

Comment:
I agree that the separation is positive.

I think the current implementation is fine, but my preference is to acquire `lock_file` in a `with` because, to me, the tradeoff of an extra level of indentation vs. guarantee that the resource will be freed, I'd prefer the guarantee. But in the current usage, it doesn't matter much because even if you forget to release the lock, the program is ending anyway so it's not a huge deal, so your call.

My other suggestion (also optional) is for `LockFile` to throw a custom error so you only need one `try`/`except` in `main`